topdown/cache/cache.go

```diff
@@ -138,6 +138,11 @@ func (c *cache) unsafeInsert(k ast.Value, v InterQueryCacheValue) (dropped int)
 		}
 	}
 
+	oldValue, oldValueOk := c.unsafeGet(k)
+	if oldValueOk {
+		c.usage -= oldValue.SizeInBytes()
+	}
+
```

Comment:
It would be better if we tried to lock the action from the caller rather than do this as described [here](https://github.com/open-policy-agent/opa/issues/5359#issuecomment-1306398327). Although this helps in representing the correct cache size, we'll be unnecessarily adding items to the list.